File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;


class RobotNewsStationNode : public rclcpp::Node   //ek class ban rhi ha jiska name ha RobotNewsStationNode aur ye ros2 ke node se inherit ho rhi h

{
    public:
       RobotNewsStationNode() : Node("robot_news_station"), robot_name_("R2D2")
       {
         publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10); 
         
         timer_ = this->create_wall_timer(500ms, std::bind(&RobotNewsStationNode::publishNews, this));
         RCLCPP_INFO(this->get_logger(), "Robot News Station has beed started");

       }
    

    private:
       void publishNews()   //ye timer ke acc call hoga
       {
          auto msg = example_interfaces::msg::String();    //ek message object banaya jiska type string h
          msg.data = std::string("Hi, this is") +robot_name_+ std::string("from the robot news");  //message me data fill kiya 
          publisher_->publish(msg); //ye message publish kr rha ha topic par

       }
       std::string robot_name_;
       rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
       rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
};

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class RobotNewsStationNode(Node):                #this is the class where we can write any thing related to node.
    def __init__(self):
        super().__init__("robot_news_station")  #parent class (node) ka constructor call karna
        self.publisher_ = self.create_publisher(String, "robot_news", 10 ) #ye publisher hai ari samaj think more about it.
        self.timer_ = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot News Station has beed started.")

    def publish_news(self):
        msg = String()
        msg.data = "Hello Nextup"
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)           #initialise the ros2 communication
    node = RobotNewsStationNode()   #start a node
    rclpy.spin(node)                #keep node spinning
    rclpy.shutdown()                #shutdown the node

if __name__ == "__main__":
    main()